scipy/spatial/tests/test_kdtree.py

```diff
@@ -216,47 +216,6 @@ def test_vectorized_query_multiple_neighbors(self):
         assert_(np.all(~np.isfinite(d[:, :, -s:])))
         assert_(np.all(i[:, :, -s:] == self.kdtree.n))
 
-    @pytest.mark.parametrize('r', [0.8, 1.1])
-    def test_single_query_all_neighbors(self, r):
-        np.random.seed(1234)
-        point = np.random.rand(self.kdtree.m)
-        with pytest.warns(DeprecationWarning, match="k=None"):
```

Comment:
Good, but we should still have a test to check that we raise a `ValidationError` in case of wrong input.

Reply:
Cool, thanks @ML-Nielsen!

The test could be quite simple.

```
def test_query_raises_for_k_none(self):
    # setup, if necessary
    with pytest.raise(ValueError, match="k must be an integer or a sequence of integers"):
        # call query with k=None
```

It can be more elaborate if there are relevant combinations to try (see `@pytest.mark.parametrize`), but this just as first inspiration.

Reply:
Nice, thanks a lot! This was a big help.

Reply:
Nice, thanks a lot! This was a big help.

Reply:
I think something went wrong with my last commit - sorry.
I wanted to add:
`def test_query_raises_for_k_none(self):
        np.random.seed(1234)

        point = np.random.rand(self.kdtree.m)
        r = 0.8
        with pytest.raises(ValueError, match="k must be an integer or a sequence of integers"):
            d, i = self.kdtree.query(point, k=None, distance_upper_bound=r)

        r = 1.1
        with pytest.raises(ValueError, match="k must be an integer or a sequence of integers"):
            d, i = self.kdtree.query(point, k=None, distance_upper_bound=r)

        query_shape = (2, 4)
        points = np.random.rand(*query_shape, self.kdtree.m)
        with pytest.raises(ValueError, match="k must be an integer or a sequence of integers"):
            d, i = self.kdtree.query(points, k=None, distance_upper_bound=r)`
but I don't think it looks right. 
Can I redo the commit?

Reply:
I think something went wrong with my last commit - sorry.
I wanted to add:
`def test_query_raises_for_k_none(self):
        np.random.seed(1234)

        point = np.random.rand(self.kdtree.m)
        r = 0.8
        with pytest.raises(ValueError, match="k must be an integer or a sequence of integers"):
            d, i = self.kdtree.query(point, k=None, distance_upper_bound=r)

        r = 1.1
        with pytest.raises(ValueError, match="k must be an integer or a sequence of integers"):
            d, i = self.kdtree.query(point, k=None, distance_upper_bound=r)

        query_shape = (2, 4)
        points = np.random.rand(*query_shape, self.kdtree.m)
        with pytest.raises(ValueError, match="k must be an integer or a sequence of integers"):
            d, i = self.kdtree.query(points, k=None, distance_upper_bound=r)`
but I don't think it looks right. 
Can I redo the commit?

Reply:
I think something went wrong with my 4th commit. That is not what I wanted to do - I wanted to commit the changes in the 5th commit. Is it possible to delete the 4th commit?

Reply:
Okay, good, thanks.